controllers/idle_handlers.go

```diff
@@ -90,27 +92,37 @@ func (r *ImageBasedUpgradeReconciler) checkManualCleanup(ctx context.Context, ib
 	return false, nil
 }
 
-//nolint:unparam
 func (r *ImageBasedUpgradeReconciler) handleFinalize(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
 	r.Log.Info("Starting handleFinalize")
 
+	r.Log.Info("Running health check for finalize (Idle) stage")
+	if err := healthcheck.HealthChecks(ctx, r.NoncachedClient, r.Log); err != nil {
+		msg := fmt.Sprintf("system must be stable before finalize (idle) stage can continue: %s", err.Error())
+		r.setHandleFinalizeFail(ibu, msg)
+		return doNotRequeue(), nil
```

Comment:
Shouldn't this be a short requeue to recheck the HealthCheck?

Reply:
Updated